sale_blanket_order/models/res_company.py

```diff
@@ -0,0 +1,7 @@
+from odoo import fields, models
+
+
+class ResCompany(models.Model):
+    _inherit = "res.company"
+
+    enable_numbered_bo = fields.Boolean("Enable numbered BO", default=False)
```

Comment:
**issue**: the name of this setting leads me to believe that without it, BOs wouldn't be numbered. Instead, this setting only changes WHEN they are numbered. A clearer name and string would help avoid possible confusion IMO. `blanket_order_seq_number_from_draft`? You can probably pick something better

(Also, when you look at res_company there are settings from all over the place, so I'd avoid abbreviations that could be unclear when taken out of context, such as `bo`)